lib/widgets/pokemon_tile_item.dart

```diff
@@ -16,7 +17,8 @@ class PokemonTileItem extends StatelessWidget {
   Widget build(BuildContext context) {
     final pokemonImageUrl = pokemon.url.toCustomUrl;
     return GestureDetector(
-      onTap: () => {},
+      /// TODO: details route will be made static in the pokemon details connector
+      onTap: () => context.go('/details', extra: pokemon),
```

Comment:
- why not use the created static const here?
- have this in a separate function i.e. `navigateToDetailsPage`

also for sub_routes I would recommend using `.push`/`.pushNamed`, since go_router is already been setup at start app, meaning routes given are already placed, then by using .push/.pushNamed you are enforcing for an specific item to be on top of the stack 🙂